### DISCUSSION TITLE
Multiphase interfacial flow

### DISCUSSION THREAD

**Discussion by MartinKlevs** [WENO-OF/WENOEXT#17] [General]
Hello,

I like the work that you're doing here.
It would be nice if there were tutorial cases for multiphase interfacial flows. I am specifically interested in using WENO to model the interface between two phases through _interFoam_.

**Comment by JanGaertner** [comment]
Hello Martin,

Have you tried using the WENO scheme and encountered any issues? What would you expect from a tutorial, e.g. what would you like to see or learn? Usage of the scheme in any solver is the same, meaning you have to inlcude the libWENOEXT.so in the system controlDict and then use the WENO scheme in the divergence operator, for the phase transport this would be then:

```
// system/controlDict
...
lib ("libWENOEXT.so");
...


// in system/fvSchemes

...
divSchemes
{
    div(phi,alpha)   Gauss WENOUpwindFit01 3; // with version dev-v3.x

}
```

Also when you are interested in transporting a volume fraction I recommend that you check out the dev-v3.x version that has a specialized WENOUpwindFit01 version which includes a limiter for 0 to 1 based on the principle described in Zhang and Su (2010). 
The limited version of WENOUpwindFit 3 1 uses then a cell limited gradient approach as known from other OpenFOAM schemes. 
 

> X. Zhang and C.-W. Shu. “On maximum-principle-satisfying high order schemes for scalar conservation laws”. In: Journal of Computational Physics 229 (9) (2010), pp. 3091–3120.

**Reply by MartinKlevs** [reply]
I have tried the scheme that you mentioned. The alpha interface seems to get smeared out over time.

Midplane slice:
![alpha](https://user-images.githubusercontent.com/31007480/113725031-9a066480-96fb-11eb-803c-196463671195.png)


 This is something that I definitely want to avoid. In some of your linked articles it is mentioned that MULES may not be the best scheme to resolve alpha. Are there any ways to ensure that the alpha boundary remains sharp and doesn't smear out over time?

**Reply by JanGaertner** [reply]
To give a bit of background information about this:
In OpenFOAM the phase transport is typically solved with the MULES solver (Multi Universal Limiter with Explicit Solution). What it does is solving the transport equation **without** diffusion in an explicit manner and then corrects the fluxes such that the values do not exceed the specified bounds, typically chosen to be 0 and 1. In the `system/fvSolution` file you can then adjust the fine tuning. For example you can set the number of correction iterations or subcycles, if you want to use substepping. 
However, a here important setting is the value `cAlpha`. cAlpha is important to select the interface compression, which keeps your interface sharp. The terms for the interface compression come from description with the velocity of center of volume, see also [1]. If `cAlpha` is set to zero you will **always** see a smearing of your interface and this has nothing to do with your scheme. Think about that the cell might be half filled with phase 1 and what you would now expect from reality is, that with time the cell gets completely filled until it spills into the next cell. However, in our numerical system it does not know that the cell is only half filled and not a mix of phase1 and phase 2. Therefore, a fraction is transported into the next cell. Just write out once the set of differential equations in 1D and you will see this directly. 
To avoid this, the trick is to use a correction velocity flux which is the drift flux coming from the formulation of center of volume. 



## References
[1] Santiago Marquez Dami´an. An Extended Mixture Model for the Simultaneous
Treatment of Short and Long Scale Interfaces. PhD thesis, UNIVERSIDAD NACIONAL DEL LITORAL, 2013.

**Reply by JanGaertner** [reply]
Maybe as a remark. If you are interested in keeping the interface sharp and maybe need to have an accurate description of such you should consider other methods than VoF e.g. PLIC. If you want to use VoF and need information about the surface wrinkling etc. and only have turbulence effects, so no phase change consider an ELSA approach.
